wdl-analysis/src/analyzer.rs

```diff
@@ -393,52 +425,73 @@ impl Analyzer {
-                    }
-
+                let metadata = path.metadata().with_context(|| {
+                    format!(
+                        "failed to read metadata for `{path}`",
+                        path = path.display()
+                    )
+                })?;
+
+                if metadata.is_file() {
+                    documents.insert(path_to_uri(&path).with_context(|| {
+                        format!(
+                            "failed to convert path `{path}` to a URI",
+                            path = path.display()
+                        )
+                    })?);
                     continue;
                 }
 
-                let pattern = format!("{path}/**/*.wdl", path = path.display());
-                let options = glob::MatchOptions {
-                    case_sensitive: true,
-                    require_literal_separator: false,
-                    require_literal_leading_dot: true,
-                };
-
-                match glob::glob_with(&pattern, options) {
-                    Ok(paths) => {
-                        for uri in paths.filter_map(|p| match p {
-                            Ok(path) => path_to_uri(&path),
-                            Err(e) => {
-                                log::error!("error while searching for WDL documents: {e}");
-                                None
-                            }
-                        }) {
-                            documents.push(uri.into());
-                        }
-                    }
-                    Err(e) => {
-                        log::error!("error while searching for WDL documents: {e}");
+                for result in WalkDir::new(&path).follow_links(true) {
+                    let entry = result.with_context(|| {
+                        format!("failed to read directory `{path}`", path = path.display())
+                    })?;
+                    if !entry.file_type().is_file()
+                        || entry.path().extension().and_then(OsStr::to_str) != Some("wdl")
+                    {
+                        continue;
                     }
+
+                    documents.insert(path_to_uri(entry.path()).with_context(|| {
+                        format!(
+                            "failed to convert path `{path}` to a URI",
+                            path = entry.path().display()
+                        )
+                    })?);
                 }
             }
 
-            documents
+            Ok(documents)
         })
-        .await
+        .await?;
+
+        if documents.is_empty() {
+            return Ok(());
+        }
+
+        // Send the add request to the queue
+        let (tx, rx) = oneshot::channel();
+        self.sender
+            .send(Request::Add(AddRequest {
+                documents,
+                completed: tx,
+            }))
+            .expect("failed to send remove request");
+
+        rx.await.unwrap_or_default();
```

Comment:
Should we do this instead? It seems like we'd want to know about a `RecvError`.

```suggestion
        rx.await?;
```

Reply:
There is also the alternative strategy that we should `.expect()` this if we expect that
it's always going to eventually resolve. In that case, that strategy applies for all of
the `unwrap_or_defaults()` I flagged here.

Reply:
This was originally `expect`, but I was running into an issue when the future was dropped (in the case of the LSP, the client canceled the request).

Let me investigate before doing so.

Reply:
We now return this as an error rather than an empty result set.